beacon-chain/blockchain/receive_attestation.go

```diff
@@ -88,30 +93,47 @@ func (s *Service) spawnProcessAttestationsRoutine(stateFeed *event.Feed) {
 		}
 
 		st := slots.NewSlotTicker(s.genesisTime, params.BeaconConfig().SecondsPerSlot)
+		pat := slots.NewSlotTickerWithOffset(s.genesisTime, -reorgLateBlockCountAttestations, params.BeaconConfig().SecondsPerSlot)
 		for {
 			select {
 			case <-s.ctx.Done():
 				return
+			case <-pat.C():
+				root := s.UpdateHead(s.ctx)
+				if err := s.forkchoiceUpdateWithExecution(s.ctx, root, s.CurrentSlot()+1); err != nil {
+					log.WithError(err).Error("could not update forkchoice")
+				}
+
 			case <-st.C():
 				if err := s.ForkChoicer().NewSlot(s.ctx, s.CurrentSlot()); err != nil {
-					log.WithError(err).Error("Could not process new slot")
+					log.WithError(err).Error("could not process new slot")
 				}
 
-				if err := s.UpdateHead(s.ctx); err != nil {
-					log.WithError(err).Error("Could not process attestations and update head")
+				root := s.UpdateHead(s.ctx)
+				s.ForkChoicer().Lock()
+				if err := s.forkchoiceUpdateWithExecution(s.ctx, root, s.CurrentSlot()); err != nil {
```

Comment:
Do we need to call `forkchoiceUpdateWithExecution` here? This is only needed if you are a proposer right?

I thought we already did this in the proposer routine

Reply:
We may need to readjust head due to attestations.